doc/contributing/submitting-changes.chapter.md

```diff
@@ -82,6 +82,50 @@ If a security fix applies to both master and a stable release then, similar to r
 
 Critical security fixes may by-pass the staging branches and be delivered directly to release branches such as `master` and `release-*`.
 
+## Deprecating/removing packages {#submitting-changes-deprecating-packages}
+
+When software projects got discontinued, are unmaintained for a year, broken for a year or have severe security vulnerabilities that don't get fixed, we should remove them. Another reason would be that there is no maintainer for the package in nixpkgs for more than 6 months. In any case, it should always be attempted to find a new maintainer or fix smaller issues first.
+
+### Steps to remove a package from nixpkgs
+
+We use jbidwatcher as an example for a discontinued project here.
+
+1. have nixpkgs checked out locally and up to date
+1. create a new branch for your change, e.g. `git checkout -b jbidwatcher`
+1. remove the actual package including it's directory, e.g. `rm -rf pkgs/applications/misc/jbidwatcher`
+1. remove the package from list of all packages (`pkgs/top-level/all-packages.nix`)
+1. add an alias for the package name in `pkgs/top-level/aliases.nix` (there is also `pkgs/misc/vim-plugins/aliases.nix` and `pkgs/top-level/python-packages.nix` for python packages)
```

Comment:
`python-packages.nix` should not keep aliases. There are some, but those should be cleaned up. A separate aliases file may be created for this functionality but does not exist yet.

Reply:
OK